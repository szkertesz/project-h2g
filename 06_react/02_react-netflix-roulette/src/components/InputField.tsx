import classes from './InputField.module.scss';

export interface IInputField {
    id: string;
    type: string;
    labelText?: string;
    onChange: React.ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement>;
    value: any;
    isFullWidth?: boolean;
    placeholder?: string;
}
const InputField: React.FC<IInputField> = ({
    id,
    type,
    labelText,
    onChange,
    value,
    isFullWidth
}) => {
    return (
        <div
            className={`
                ${classes['input-field']}
                ${isFullWidth ?
                    classes['input-field--full'] :
                    ''
                }
            `}
        >
            <label htmlFor={id}>{labelText}</label>
            <input
                type={type}
                id={id}
                name={id}
                onChange={onChange}
                value={value}
            />
        </div>
    );
};

export default InputField