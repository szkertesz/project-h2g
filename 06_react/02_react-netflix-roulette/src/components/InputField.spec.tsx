import React from 'react';
import InputField from './InputField';
import { render, screen } from '@testing-library/react';

describe('GenreList', () => {
    describe('isFullWidth is true', () => {
        const isFullWidth = true
        it('should render component', () => {
            const id ='test id';
            const type = 'text';
            const labelText = 'label text';
            const value = 3;

            const { container } = render(<InputField id={id} type={type} labelText={labelText} onChange={jest.fn()} value={value} isFullWidth={isFullWidth}/>);
            expect(container).toMatchSnapshot();
        });
    });
    describe('isFullWidth is false', () => {
        const isFullWidth = false
        it('should render component', () => {
            const id ='test id';
            const type = 'text';
            const labelText = 'label text';
            const value = 3;

            const { container } = render(<InputField id={id} type={type} labelText={labelText} onChange={jest.fn()} value={value} isFullWidth={isFullWidth}/>);
            expect(container).toMatchSnapshot();
        });
    });

});
