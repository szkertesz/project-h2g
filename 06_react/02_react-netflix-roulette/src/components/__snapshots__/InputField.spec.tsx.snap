// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreList isFullWidth is false should render component 1`] = `
<div>
  <div
    class="
                input-field
                
            "
  >
    <label
      for="test id"
    >
      label text
    </label>
    <input
      id="test id"
      name="test id"
      type="text"
      value="3"
    />
  </div>
</div>
`;

exports[`GenreList isFullWidth is true should render component 1`] = `
<div>
  <div
    class="
                input-field
                input-field--full
            "
  >
    <label
      for="test id"
    >
      label text
    </label>
    <input
      id="test id"
      name="test id"
      type="text"
      value="3"
    />
  </div>
</div>
`;
